src/trg-prefs.c

```diff
@@ -568,7 +568,7 @@ void trg_prefs_load(TrgPrefs * p)
                                                 TRG_PREFS_KEY_PROFILES);
     }
 
-    n_profiles = json_array_get_length(profiles);
+    n_profiles = (gint)json_array_get_length(profiles);
```

Comment:
I wonder if this is right... this would be undefined behavior if `n_profiles > MAX_INT` I guess.

Reply:
`profile_id` might be safer to cast to `guint` as it can't be negative (or that would be a bug in trg).